### PR TITLE
Dev/max statement update

### PR DESCRIPTION
**Pull request by RachelErin**

### CODE REVIEW

**Review by jdebaets — APPROVED** [review]

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorParser.java [hunk 1]
```diff
@@ -91,20 +93,36 @@ public SourceMonitorReport invoke(java.io.File workspace, VirtualChannel channel
...
             List<?> metricsEltList = metricsElt.getChildren();
+            Element functionMetrics = checkpoint.getChild("function_metrics");
+            List<?> functionMetricsEltList = functionMetrics.getChildren("function");
+
             for (int j = 0; j < metricsEltList.size(); j++) {
                 Element metricElt = (Element)metricsEltList.get(j);
                 metricsSummaryMap.put(metricNameMap.get(metricElt.getAttributeValue("id")), metricElt.getValue());
             }
+            int numFunctions = Integer.parseInt(functionMetrics.getAttributeValue("function_count"));
+            for (int k = 0; k < numFunctions; k++){
```
Comment (greghoeppner) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorParser.java]: The function is getting pretty complex. Maybe break out the new code into a separate function.

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorResult.java [hunk 2]
```diff
@@ -49,7 +49,19 @@ public SourceMonitorReport getReport(){
         return owner;
     }
 
-
+    public String getDetailedMetrics(){
+        StringBuilder builder = new StringBuilder();
+        for (int i=0; i<report.getDetailedMetrics().size();i++){
+            builder.append("<tr><td>");
```
Comment (greghoeppner) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorResult.java:55]: What happens when you load an report where detailed metrics has not been set? Do you get a blank table or a null exception?

Comment (RachelErin) [reply src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/SourceMonitorResult.java:55]: Thanks. Added if in jelly file that works with old/new.

File: src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FunctionStats.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package com.thalesgroup.hudson.plugins.sourcemonitor;
+
+import java.io.Serializable;
```
Comment (kaneofawesome) [inline src/main/java/com/thalesgroup/hudson/plugins/sourcemonitor/FunctionStats.java:3]: maybe a comment why is appropriate here

**Review by kaneofawesome — APPROVED** [review]

**Review by greghoeppner — APPROVED** [review]

**Pull request merged**